File: src/main/scala/BroadCastTest.scala
package main.scala

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object BroadCastTest {
   def main(args: Array[String]): Unit = {
        val conf = new SparkConf().setAppName("AggregateOperator").setMaster("local[3]")
        val sc = new SparkContext(conf)
        val userNames = Array(
            "xuruyun",    
            "zhouzhilei",
            "panglei",
            "laochen"
        )
        
        val userNameRDD = sc.parallelize(userNames)
       
        val blackNameBroadcaste = sc.broadcast("xuruyun")
        
        val linesCountAccumulator = sc.accumulator(0)
        val filteredUserNameRDD = userNameRDD.filter {x =>{
        	linesCountAccumulator.add(1)
          !blackNameBroadcaste.value.equals(x)
        }}
        
        filteredUserNameRDD.collect().foreach(println)
        println(linesCountAccumulator.value)
        
   }
}

File: src/main/java/broadcast/BroadCastTest.java
package main.java.broadcast;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.broadcast.Broadcast;

public class BroadCastTest {
	public static void main(String[] args) {
		
		SparkConf sparkConf = new SparkConf()
				.setMaster("local")
				.setAppName("BroadCastTest");
		JavaSparkContext sc = new JavaSparkContext(sparkConf);
		List<String> userNames = Arrays.asList(
				"xuruyun",
				"laochen",
				"liangyongqi",
				"xiaolaoshi",
				"panglaoshi"
				);
		JavaRDD<String> userNameRDD = sc.parallelize(userNames);
		
		/**
		 * 广播变量必须在Driver端定义 
		 */
		final Broadcast<String> blackNameBroadcast = sc.broadcast("xuruyun");
		
		
		JavaRDD<String> filteredUserNameRDD = userNameRDD.filter(new Function<String, Boolean>() {
			/**
			 * 
			 */
			private static final long serialVersionUID = 1L;

			@Override
			public Boolean call(String userName) throws Exception {
				String blackName = blackNameBroadcast.value();
				return !blackName.equals(userName);
			}
		});
				
		List<String> collect = filteredUserNameRDD.collect();
		for (String string : collect) {
			System.out.println(string);
		}
		
		sc.stop();
	}
}
